src/zkevm_specs/evm/execution/mulmod.py

```diff
@@ -0,0 +1,75 @@
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+from zkevm_specs.util import FQ, RLC
+
+# TODO Copied from ./addmod.py
+# Returns 1 when a is lower than b, 0 otherwise
+def lt_u256(instruction: Instruction, a: RLC, b: RLC) -> FQ:
+    # decode RLC to bytes for a and b
+    a8s = a.le_bytes
+    b8s = b.le_bytes
+
+    a_lo = instruction.bytes_to_fq(a8s[:16])
+    a_hi = instruction.bytes_to_fq(a8s[16:])
+    b_lo = instruction.bytes_to_fq(b8s[:16])
+    b_hi = instruction.bytes_to_fq(b8s[16:])
+
+    a_lt_b_lo, _ = instruction.compare(a_lo, b_lo, 16)
+    a_lt_b_hi, a_eq_b_hi = instruction.compare(a_hi, b_hi, 16)
+
+    a_lt_b = instruction.select(
+        a_lt_b_hi, FQ(1), instruction.select(a_eq_b_hi * a_lt_b_lo, FQ(1), FQ(0))
+    )
+
+    return a_lt_b
+
+
+def mulmod(instruction: Instruction):
+
+    MOD = 2**256
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.MULMOD)
+
+    a = instruction.stack_pop()
+    b = instruction.stack_pop()
+    n = instruction.stack_pop()
+    pushed_r = instruction.stack_push()
+
+    if n.int_value == 0:
+        a_reduced = a.int_value
+        k1 = 0
+        r = RLC((a.int_value * b.int_value) % MOD)
+        k2 = 0
+    else:
+        a_reduced = a.int_value % n.int_value
+        k1 = a.int_value // n.int_value
+        r = pushed_r
+        k2 = (a_reduced * b.int_value) // n.int_value
+
+    a_reduced_times_b = a_reduced * b.int_value
+    e = RLC(a_reduced_times_b % MOD)
+    d = RLC(a_reduced_times_b // MOD)
+
+    # Safety check
+    assert (a_reduced_times_b) == k2 * n.int_value + r.int_value
+
+    # Reduction of first factor
+    instruction.mul_add_words(RLC(k1), n, RLC(a_reduced), a)
```

Comment:
Here what we want is:
`a_reduced = a mod n`
and we do it via
`k1 * n + a_reduced = a mod 8**256`

The first thing that comes to my mind is that we should constrain
`a_reduced < n`.  Otherwise there can be many pairs of `(k1, a_reduced)` which are valid, right?

The second thing I'm wondering is: can this multiplicacion (or the addition) overflow in the 256 bits?

Reply:
Yes, `a_reduced < n` constraint is missing. I'll add that!
Regarding the second point, we don't need to worry about overflow in this equation. `a <= MAXU256` so neither side of the equation will overflow.

Reply:
You're right! I'll add this too